generic_parser/entrypoint_parser.py

```diff
@@ -547,51 +589,71 @@ def add_to_arguments(args, entry_params=None, **kwargs):
     return args
 
 
+# parameter adders ---
+
 def add_params_to_generic(parser, params):
     """ Adds entry-point style parameter to either
     ArgumentParser, DictParser or EntryPointParameters
     """
-    params = copy.deepcopy(params)
-    params = dict2list_param(params)
 
     if isinstance(parser, EntryPointParameters):
-        for param in params:
-            parser.add_parameter(**param)
+        parser = _add_params_to_entrypoint_parser(parser, params)
 
     elif isinstance(parser, ArgumentParser):
```

Comment:
group with the previous statement?

Reply:
what?

Reply:
The previous `if` and this `elif` lead to the exact same action.
Should be `if ... or ...`